airflow/utils/dag_processing.py

```diff
@@ -1193,13 +1195,11 @@ def heartbeat(self):
 
             self._file_path_queue.extend(files_paths_to_queue)
 
-        zombies = self._find_zombies()
-
         # Start more processors if we have enough slots and files to process
         while (self._parallelism - len(self._processors) > 0 and
                len(self._file_path_queue) > 0):
             file_path = self._file_path_queue.pop(0)
-            processor = self._processor_factory(file_path, zombies)
+            processor = self._processor_factory(file_path, self._zombies)
```

Comment:
I wondered if the list should be copied, but it's fine because `_find_zombies()` creates a new list and swaps `self._zombies`.

Reply:
Since we're using multiprocessing, the list will always be copied as they won't be able to share any object. Even it shares it's still fine since we only read from it.

A quick demo on the no share part:

```
root@7949bc1f5c5c:/opt/airflow# python test.py
['123']
root@7949bc1f5c5c:/opt/airflow# cat test.py
import multiprocessing
from pprint import pprint

def worker(the_list):
    """worker function"""
    the_list.append(multiprocessing.current_process().name)
    return

if __name__ == '__main__':
    jobs = []
    the_list = ['123']
    for i in range(3):
        p = multiprocessing.Process(target=worker, args=(the_list, ), name="subprocess-{}".format(i))
        jobs.append(p)
        p.start()
    pprint(the_list)
```